src/core/AutoRest.Core/Settings.cs

```diff
@@ -81,6 +82,13 @@ public Settings()
         public string Input { get; set; }
 
         /// <summary>
+        /// Gets or sets a name for the generated client models Namespace and Models output folder
+        /// </summary>
+        [SettingsInfo("Name to use for the generated client models namespace and folder name. Not supported by all code generators.")]
+        [SettingsAlias("mname")]
+        public string ModelsName { get; set; }
```

Comment:
And if this was the whole namespace (as opposed to just the suffix) then you wouldn't have to do string smashing down below...